tests/chez/chez015/Numbers.idr

```diff
@@ -9,7 +9,7 @@ large : (a: Type) -> a
 large Integer = 3518437212345678901234567890123
 -- int close to 2ˆ63
 -- we expect some operations will overflow
-large Int = 3518437212345678901234567890
+large Int = 9223372036854775800
```

Comment:
It's unfortunate that you are changing the test case at the same time as you are refactoring
because now we can't tell whether the refactoring is innocuous.

Reply:
Well, I didn't like this test case for two reasons:

- the original `Int` is nowhere near 2^63, so either the comment or the literal is false
- the original `Int` is out of bounds. From #1101 we know, that right now Idris2 will not truncate this as it should, therefore is hard to figure out, what the point of this test was in the first place. Eventually (when we fix #1101), this needs to be changed anyway, doesn't it?

The new test case still makes sure that truncation behaves correctly. Still, if you think we should go back to the old test, I'll revert this change.

Reply:
OK, so here's the problem with this test case: It looks like this refactoring already fixes #1101, since the following line, which did not truncate when going from `Integer` to `Int` was replaced with the call to `castInt`, which does proper truncation for all combinations of integer types:

```idris
schOp (Cast IntegerType IntType) [x] = x
```

But with this change, the old version of the test fails, and it is the calls to `div` and `mod` that fail since

```
(a `mod` b) `mod` c/= (a `mod` c) `mod` b
```

in the general case.